test/extensions/filters/http/wasm/test_data/grpc_callout_cpp.cc

```diff
@@ -0,0 +1,51 @@
+// NOLINT(namespace-envoy)
+#include <string>
+#include <unordered_map>
+#include "proxy_wasm_intrinsics.h"
+#include "proxy_wasm_intrinsics_lite.pb.h"
+
+class ExampleContext : public Context {
+ public:
+  explicit ExampleContext(uint32_t id, RootContext *root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders() override;
+};
+
+static RegisterContextFactory
+    register_ExampleContext(CONTEXT_FACTORY(ExampleContext));
+
+class MyGrpcCallHandler : public GrpcCallHandler<google::protobuf::Value> {
+ public:
+  MyGrpcCallHandler(ExampleContext *context)
+      : GrpcCallHandler<google::protobuf::Value>(context), context_(context) {}
+  void onCreateInitialMetadata() override {}
+  void onSuccess(google::protobuf::Value &&response) override {
+    logInfo(response.string_value());
+    continueRequest();
+  }
+  void onFailure(GrpcStatus status,
+                 std::unique_ptr<WasmData> error_message) override {
+    logInfo(std::string("failure ") + std::to_string(static_cast<int>(status)) +
+            std::string(error_message->view()));
+
+    // TODO wasm engine must support fail closed: expose abortRequest() or similar
+    continueRequest();
```

Comment:
This test case passes because of this continueRequest().  There should be some mechanism to abort the downstream request onFailure.

Reply:
Do we have an issue for this?

Reply:
https://github.com/envoyproxy/envoy-wasm/issues/74